contracts/external/cover/CoverBroker.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/security/ReentrancyGuard.sol";
+
+import "../../interfaces/ICover.sol";
+import "../../interfaces/ICoverBroker.sol";
+import "../../interfaces/IMemberRoles.sol";
+
+// * the contract is kyc-ed and whitelisted as a member
+// * interacts with the Nexus Cover contract on behalf of the user
+// * mints the NFT in the end user's wallet
+
+// Functions
+// * buyCover: calls Cover.buyCover and pass end user's address as BuyCoverParams.owner, so the NFT gets minted directly in their wallet
+// * switchMembership: in case we want in the future to redeploy the contract. This can be called by an owner address only
+// * transferFunds (?) - fish any tokens that accidentally get sent to the contract - can only be called by the owner address above
+
+// Further notes:
+
+// * buying using ETH, DAI or USDC
+// * user -> external contract -> buy cover (will have extra buffer - so it does not revert)
+//* need to make sure any extra monies is sent back to user
+// * any left money before buyCover should stay in contract
+
+contract CoverBroker is ICoverBroker, ReentrancyGuard {
+  // Immutables
+  address owner;
+  ICover cover;
+  IMemberRoles memberRoles;
+
+  // Constants
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+  modifier onlyOwner() {
+    if (msg.sender != owner) {
+      revert OnlyOwner();
+    }
+    _;
+  }
+
+  constructor(address _owner, address _cover, address _memberRoles) {
+    owner = _owner;
+    cover = ICover(_cover);
+    memberRoles = IMemberRoles(_memberRoles);
+  }
+
+  function buyCover(
+    BuyCoverParams memory params,
+    PoolAllocationRequest[] calldata poolAllocationRequests
+  ) external payable nonReentrant returns (uint coverId) {
+    uint ethBalanceBefore = address(this).balance;
+    // set the owner to the caller
+    params.owner = msg.sender;
+    // call cover.buyCover with msg.value and params
+    coverId = cover.buyCover{value: msg.value}(params, poolAllocationRequests);
+    // send back any ETH refund to user
+    uint ethRefund = address(this).balance - ethBalanceBefore;
```

Comment:
i'm forwarding whatever the value the caller sends to this method to buyCover.

if the cover turned out cheaper than `msg.value`, I'm assuming the Cover contract will send the "change" back to CoverBroker.

is this assumption correct?